chef.py

```diff
@@ -300,118 +302,6 @@ def truncate_metadata(data_string):
-
-            filename = derive_filename(url)
-            node[attr] = filename
-
-            print("        Downloading", url, "to filename", filename)
-            download_file(url, destination, request_fn=make_request,
-                    filename=filename, middleware_callbacks=content_middleware)
-
-    def js_middleware(content, url, **kwargs):
-        # Download all images referenced in JS files
-        for img in IMAGES_IN_JS_RE.findall(content):
-            url = make_fully_qualified_url('/images/%s' % img)
-            print("        Downloading", url, "to filename", img)
-            download_file(url, destination, subpath="images",
-                    request_fn=make_request, filename=img)
-
-        # Polyfill localStorage and document.cookie as iframes can't access
-        # them
-        return (content
-            .replace("localStorage", "_localStorage")
-            .replace('document.cookie.split', '"".split')
-            .replace('document.cookie', 'window._document_cookie'))
-
-    def css_node_filter(node):
-        return "stylesheet" in node["rel"]
-
-    def css_content_middleware(content, url, **kwargs):
-        file_dir = os.path.dirname(urlparse(url).path)
-
-        # Download linked fonts and images
-        def repl(match):
-            src = match.group(1)
-            if src.startswith('//localhost'):
-                return 'url()'
-            # Don't download data: files
-            if src.startswith('data:'):
-                return match.group(0)
-            src_url = make_fully_qualified_url(os.path.join(file_dir, src))
-            derived_filename = derive_filename(src_url)
-            download_file(src_url, destination, request_fn=make_request,
-                    filename=derived_filename)
-            return 'url("%s")' % derived_filename
-
-        return CSS_URL_RE.sub(repl, content)
-
-    # Download all linked static assets.
-    download_assets("img[src]", "src")  # Images
-    download_assets("link[href]", "href",
-            content_middleware=css_content_middleware,
-            node_filter=css_node_filter)  # CSS
-    download_assets("script[src]", "src", content_middleware=js_middleware) # JS
-    download_assets("source[src]", "src") # Audio
-    download_assets("source[srcset]", "srcset") # Audio
-
-    # ... and also run the middleware on CSS/JS embedded in the page source to
-    # get linked files.
-    for node in doc.select('style'):
-        node.string = css_content_middleware(node.get_text(), url='')
-
-    for node in doc.select('script'):
-        if not node.attrs.get('src'):
-            node.string = js_middleware(node.get_text(), url='')
-
-    return doc
-
-
-url_blacklist = [
-    'analytics.js',
-]
-
-def is_blacklisted(url):
-    return any((item in url) for item in url_blacklist)
-
-
-def derive_filename(url):
-    return "%s.%s" % (uuid.uuid4().hex, os.path.basename(urlparse(url).path))
-
-
 class Dummy404ResponseObject(requests.Response):
```

Comment:
You won't need here anymore.

Also please update requirements.txt and then merge away!

Reply:
Removed and updated! Thanks!